### ISSUE TITLE
Failed to Login to Peloton - Causes Crash

### ISSUE THREAD

**Issue by philosowaffle** [philosowaffle/peloton-to-garmin#67]
Seems that Peloton has shut down the previous endpoint this tool used to sign in.  Will take some work to find a replacement endpoint.... if one even exists.

```
2021-01-25 12:55:57,750 - DEBUG - peloton-to-garmin.PelotonApi: <Response [404]>
```

```
File "peloton-to-garmin.py", line 95, in <module>
    PelotonToGarmin.run(config)
  File "peloton-to-garmin.py", line 27, in run
    pelotonClient = pelotonApi.PelotonApi(config.peloton_email, config.peloton_password)
  File "G:\Code\Git\peloton-to-garmin\lib\pelotonApi.py", line 25, in __init__
    parsed_response = util.parse_response(response)
  File "G:\Code\Git\peloton-to-garmin\lib\util.py", line 21, in parse_response
    parsed_response = json.loads(response.text)
  File "C:\Users\Bailey\AppData\Local\Programs\Python\Python36-32\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Users\Bailey\AppData\Local\Programs\Python\Python36-32\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Users\Bailey\AppData\Local\Programs\Python\Python36-32\lib\json\decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by philosowaffle** [comment]
original authentication strategy was taken from here: https://github.com/geudrik/peloton-client-library/blob/master/API_DOCS.md

**Comment by FredB76** [comment]
"Export original" from Strava produces a .tcx that needs a few small fixes before it can be imported in Garmin Connect. Maybe there's a way from there?

**Comment by philosowaffle** [comment]
Not a bad backup plan, can fall back to strava for getting the data if cant find another authentication option. Downside is it would require everyone to have a strava account and link it to their peloton account.

**Comment by eRaid6** [comment]
It seems that Peloton has a official relationship with Strava, unlike Garmin, so having users setup their Peloton account to sync to Strava and then having the script pull from Strava and upload to Garmin might be a good long term strategy.    Just my $0.02.

Thanks for this awesome project, I've been using it for the last year and love it.

**Comment by philosowaffle** [comment]
Will definitely look into switching over to strava.  I have also reached out to another peloton project [here](https://github.com/geudrik/peloton-client-library/issues/12) to see if they can help out.  Replacing the url with a new login url would be a 5min change vs. a rewrite to fetch from strava and sync which I probably couldn't work on until this weekend.

**Comment by philosowaffle** [comment]
Found the 5min fix I was hoping for.  This is now resolved in release v1.0.4 

https://github.com/philosowaffle/peloton-to-garmin/releases/tag/v1.0.4
